app/script/view_model/bindings/CommonBindings.js

```diff
@@ -490,52 +490,63 @@ ko.bindingHandlers.removed_from_view = {
  * Element is in viewport. return true within the callback to dispose the subscription
  */
 ko.bindingHandlers.in_viewport = (function() {
-  const listeners = [];
-
-  // listeners can be deleted during iteration
-  const notifyListeners = _.throttle(event => {
-    for (let index = listeners.length; index--; ) {
-      listeners[index](event);
-    }
-  }, 300);
-
-  window.addEventListener('scroll', notifyListeners, true);
-  notifyListeners();
+  const overlayedElements = new Map();
+
+  const isOverlayed = domElement => {
+    const box = domElement.getBoundingClientRect();
+    const elementAtPoint = document.elementFromPoint(box.x, box.y);
+    return (
+      elementAtPoint &&
+      !elementAtPoint.contains(domElement) &&
+      !domElement.contains(elementAtPoint) &&
+      domElement !== elementAtPoint
+    );
+  };
 
-  return {
-    init(element, valueAccessor, allBindingsAccessor) {
-      function _inView(domElement) {
-        const box = domElement.getBoundingClientRect();
-        const isInRightPanel = document.querySelector('#right-column').contains(domElement);
-        return (
-          box.right >= 0 &&
-          box.bottom >= 0 &&
-          (isInRightPanel || box.left <= document.documentElement.clientWidth) &&
-          box.top <= document.documentElement.clientHeight
-        );
+  const checkOverlayedElements = mutations => {
+    mutations.forEach(({removedNodes}) => {
+      if (removedNodes && removedNodes.length) {
+        overlayedElements.forEach((onVisible, element) => {
+          if (!isOverlayed(element)) {
+            onVisible();
+            removeVisibleListener(element);
+          }
+        });
       }
+    });
+  };
 
-      function _dispose() {
-        z.util.ArrayUtil.removeElement(listeners, _checkElement);
-      }
+  const mutationObserver = new MutationObserver(checkOverlayedElements);
```

Comment:
@gregor I was thinking, probably we can extract this to a UI util somewhere with a nice and clean API (like you did with the IntersectionObserver). But then I am having troubles with naming it. Is it a 
`PaintedObserver` (checks that the element is painted on screen and not hidden behind another
`OverlayedObserver`
`VisibleObserver` (which might not be absolutely true, not sure yet)